### ISSUE TITLE
Video player component

### ISSUE THREAD

**Issue by CarbonPool** [slint-ui/slint#2816]
Whether to support playing video, or other ways to achieve

**Comment by tronical** [comment]
I'm not sure that this should be a core feature of Slint - creating a multimedia framework feels out of scope. But we want users to be able to use existing frameworks. So we made an example that demonstrates how to use ffmpeg to play video with Slint: https://github.com/slint-ui/slint/tree/master/examples/ffmpeg

**Comment by felix307253927** [comment]
@tronical I think video or audio components are necessary as they are used in many scenarios

**Comment by tronical** [comment]
Yes, at the same time the choice of framework is a highly opinionated area. If Slint chooses one framework, it will likely not end up being the preferred choice of others. So my preference is to have enablers in Slint and supporting examples and documentation that empowers everyone to make their choices. For cross-platform I think ffmpeg works best, but for embedded Linux often vendors specifically support gstreamer and thus becomes the best choice.

**Comment by felix307253927** [comment]
The performance of the above example is relatively poor. Are there any examples of integrating MPV or VLC

**Comment by tronical** [comment]
Right, the current ffmpeg example is not using hardware acceleration yet. I've toyed with the idea of adding support for rendering vaapi yuv textures via OES_external_image for Linux, and VideoToolBox on macOS.

**Comment by felix307253927** [comment]
I am currently using QT in the project and using VLC-QT to play videos. My team is relatively weak in C++, and I am planning to try Slint UI, but I don't know how to play videos in Slint UI

**Comment by tronical** [comment]
I see. Yeah, we also need a C++ example. libvlc or perhaps [placebo](https://code.videolan.org/videolan/libplacebo) would be a good candidate for that, to do similar things as VLC-Qt does. In any case, I agree that we should add something to Slint. My preference is by example, but we could also have a component that uses Vlc, one for ffmpeg, etc.

**Comment by felix307253927** [comment]
That's great. I'm really looking forward to it

**Comment by mcirsta** [comment]
This was an interesting discussion. I am planning to use SLIT for an embedded UI. We will need to play some media too ( short clips ) and hardware acceleration will be needed, most likely everything will be on top of GStreamer.

**Comment by NateAGeek** [comment]
I'm currently working on a project that needs video. I kinda mashed together the ffmpeg and opengl projects together. To handle any video larger than 640p I noticed SWScale was really taking up a lot of resources. I did make a shader to work with native YUV... But kinda think I might want to switch to VLClib or something that handles a lot of the underlying sampling and decoding. 

[SenToAni](https://github.com/NateAGeek/SenToAni) if anyone is interested in the code.
